internal/commands/push.go

```diff
@@ -319,3 +332,14 @@ func (r *plainDisplay) onEvent(ev remotes.FixupEvent) {
 		}
 	}
 }
+
+func checkFlags(flags *pflag.FlagSet) error {
+	allPlatforms, err := flags.GetBool("all-platforms")
```

Comment:
You could also add `opts pushOptions` as a parameter and check directly `opts.allPlatforms` ?
```go
if opts.allPlatforms && flags.Changed("platform"){
    return fmt.Errorf("--all-plaforms and --plaform flags cannot be used at the same time")
}
return nil
```

Reply:
Done